lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1762,6 +1767,14 @@ def _get_file_hash(b_path, filename):  # type: (bytes, str) -> str
         return _consume_file(fp)
 
 
+def remove_or_unlink_dir(b_path: bytes):
+    if os.path.islink(b_path):
+        if os.path.exists(os.path.realpath(b_path)):
```

Comment:
Does this mean that if a symlink is pointing to a non-existing path, its removal is skipped? Why?

Reply:
It hasn't been implemented yet. We call `shutil.rmtree` on devel without checking the type, which is a traceback for files too. Either the error needs to be fixed, or we should remove non-directories also.

Reply:
Updated to remove non-directories too, here's the current behavior https://github.com/ansible/ansible/pull/81355/files#diff-b70502974d601e8bb20fb73c347f4ac5ba4dc6c6823d787d6e9e85f5ea6f056fR89-R108.

Reply:
Great! Marking the thread as resolved.

Reply:
It's not resolved. It's still a draft because I thought the behavior needs to be discussed (for example, compare with roles).

Reply:
I [tried to explain](https://github.com/ansible/ansible/pull/81355#discussion_r1277561302), as your point above was the main reason this was a WIP in the first place (what to do with non-collections specifically). An arbitrary demand was made, maybe without considering the whole picture, or what I was stuck on, and I implemented it for discussion after explaining why it's not obviously the right behavior.

Reply:
I see. Perhaps I didn't understand the entire context. Thanks for pointing this out. This PR poped up in my inbox the other day so I tried looking into what's changed since I saw it last but I'd already forgotten all the considerations.